google/cloud/storage/internal/grpc_configure_client_context.cc

```diff
@@ -29,10 +29,13 @@ void ApplyRoutingHeaders(
 
 void ApplyRoutingHeaders(grpc::ClientContext& context,
                          storage::internal::UploadChunkRequest const& request) {
-  static auto* bucket_regex =
+  static auto* slash_format =
       new std::regex{"(projects/[^/]+/buckets/[^/]+)/.*", std::regex::optimize};
-  std::smatch match;
-  if (std::regex_match(request.upload_session_url(), match, *bucket_regex)) {
+  static auto* colon_format =
+      new std::regex{"(projects/[^/]+/buckets/[^:]+):.*", std::regex::optimize};
+  for (auto const* re : {slash_format, colon_format}) {
```

Comment:
(due dilligence) why do we need two formats?

I only see `UploadChunk` calling `rpc WriteObject`, which doesn't / can't have routing annotations. I don't remember the source of this regex.

Reply:
The service is going to change from format 1 to format 2.  I don't want to have to perfectly coordinate which format works.

Reply:
Also `WriteObject()` lacks the annotations because it is a streaming RPC. That is why we have to manually insert them too.